OlinkAnalyze/vignettes/bridging_introduction.Rmd

```diff
@@ -0,0 +1,233 @@
+assay between the bridging samples as adjustment factor then use these adjustment 
+factors to adjust NPX values between two datasets. The output from 
+`olink_normalization()` function is a NPX table with adjusted NPX value in the 
+column `NPX`.
+
+```{r bridging, message=FALSE}
+# Find overlapping samples
+npx_1 <- npx_data1 %>% 
+  mutate(dataset = "data1")
+npx_2 <- npx_data2 %>% 
+  mutate(dataset = "data2")
+
+overlap_samples <- intersect(npx_1$SampleID, npx_2$SampleID) %>% 
+  data.frame() %>% 
+  filter(!str_detect(., 'CONTROL_SAMPLE')) %>% #Remove control samples
+  pull(.)
+# Perform Bridging normalization
+npx_br_data <- olink_normalization(df1 = npx_1, 
+                                   df2 = npx_2, 
+                                   overlapping_samples_df1 = overlap_samples,
+                                   df1_project_nr = '20200001',
+                                   df2_project_nr = '20200002',
+                                   reference_project = '20200001')
+glimpse(npx_br_data)
+
+```
+
+## Perform evaluation of bridging normalization
+
+
+First, check NPX distribution in datasets after bridging normalization.
+
+```{r densitybr, message=FALSE}
+# Plot NPX density after bridging normalization
+
+npx_br_data %>% 
+  mutate(Panel = gsub("Olink ", "", Panel)) %>% 
+  ggplot(aes(x = NPX, fill = Panel)) +
+  geom_density(scale = 1, alpha = 0.7) + 
+  facet_grid(dataset~Panel) +
+  olink_fill_discrete(coloroption = c('teal', 'lightblue')) +
+  set_plot_theme() + 
+  ggtitle("After bridging normalization: NPX distribution") +
+  theme(axis.title.x = element_blank(),
+        axis.title.y = element_blank(),
+        strip.text = element_text(size=16),
+        legend.title = element_blank(),
+        legend.position = "none")
+
+``` 
+
+Then summarize number of assays that have adjustment factors in certain ranges.
+
+```{r tbl_adj_fct, message=FALSE}
+breaks <- c(-Inf, -1, -0.25, 0.25, 1, Inf)
+labels <- c("[< -1]", "(-1,-0.25]", "(-0.25,0.25]", "(0.25,1]", "(>1]")
+
+reference_project <- '20200001' # reference project defined in bridging normalization
+npx_br_data %>% 
+  mutate(Panel = gsub("Olink ", "", Panel)) %>% 
+  filter(Project != reference_project) %>% 
+  mutate(group = cut(Adj_factor, breaks = breaks, 
+            labels = labels)) %>% 
+  select(OlinkID, Adj_factor, Panel, group) %>% 
+  distinct() %>% 
+  group_by(Panel, group) %>% tally() %>% 
+  tidyr::pivot_wider(names_from = group, values_from = n) %>% 
+  tibble::column_to_rownames(var = "Panel") %>% 
+  relocate(any_of(labels)) %>% 
+  kbl(booktabs = T,
+      digits = 2,
+      caption = "Distribution of assays in different ranges of adjustment factors") %>%
+  kable_styling(bootstrap_options = "striped", full_width = F, position = "center",
+                latex_options = "HOLD_position")
+
+``` 
+
+
+Finally, use PCA plot to check whether bridging normalization has effect in correcting batch effects.
+
```

Comment:
Add comment on how we see in the PCA that bridging normalization had effect in correcting batch effects.